src/NLog/Targets/NetworkTarget.cs

```diff
@@ -289,12 +291,17 @@ protected override void Write(AsyncLogEventInfo logEvent)
 
             byte[] bytes = GetBytesToWrite(logEvent.LogEvent);
 
-            if (KeepConnection)
+            var networkTargetOverflowAction = OnOverflow.ToValueSafe(logEvent);
```

Comment:
@snakefoot please your opinion on `ToValueSafe`. I don't like it.

But otherwise, I had null refs all the time.

I can't make `Layout<T>` as `struct` I guess - and also not sure if that would be a good idea if that was even possible

Reply:
idea:

- hide ToValue with explicit interface implementation, and rename to `.GetValue`. Need an extra interface for that.
- rename `ToValueSafe` to `GetValueOrDefault`

Reply:
@snakefoot any ideas on this?

Reply:
Think the extension-method is a good way to handle that `Layout<T>` can be null. Maybe call it `RenderValue` and let the default value be optional-parameter so it is visible for the user. Should make it easy to guess what what will happen if null or unknown string-value.

Reply:
Thanks!